neptune-core/src/application/json_rpc/core/model/wallet/mutator_set.rs

```diff
@@ -4,36 +4,75 @@ use serde::Deserialize;
 use serde::Serialize;
 use tasm_lib::prelude::Digest;
 use tasm_lib::triton_vm::prelude::BFieldElement;
+use tasm_lib::twenty_first::prelude::MmrMembershipProof;
 
 use crate::application::json_rpc::core::model::block::body::RpcMutatorSetAccumulator;
 use crate::application::json_rpc::core::model::block::transaction_kernel::RpcChunkDictionary;
 use crate::util_types::mutator_set::archival_mutator_set::IndexedAoclAuthPath;
 use crate::util_types::mutator_set::archival_mutator_set::MsMembershipProofPrivacyPreserving;
+use crate::util_types::mutator_set::ms_membership_proof::MsMembershipProof;
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Default)]
-pub struct RpcIndexedAoclAuthPaths(pub BTreeMap<u64, Vec<Digest>>);
+pub struct RpcMmrMembershipProof(Vec<Digest>);
+
+impl From<MmrMembershipProof> for RpcMmrMembershipProof {
+    fn from(value: MmrMembershipProof) -> Self {
+        Self(value.authentication_path)
+    }
+}
+
+impl From<RpcMmrMembershipProof> for MmrMembershipProof {
+    fn from(value: RpcMmrMembershipProof) -> Self {
+        MmrMembershipProof::new(value.0)
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Default)]
+pub struct RpcIndexedAoclAuthPaths(pub BTreeMap<u64, RpcMmrMembershipProof>);
 
 impl From<Vec<IndexedAoclAuthPath>> for RpcIndexedAoclAuthPaths {
     fn from(value: Vec<IndexedAoclAuthPath>) -> Self {
-        RpcIndexedAoclAuthPaths(
+        Self(
             value
                 .into_iter()
-                .map(|indexed| (indexed.leaf_index, indexed.auth_path.authentication_path)) // TODO: RpcMmrMembershipProof
+                .map(|indexed| (indexed.leaf_index, indexed.auth_path.into()))
                 .collect(),
         )
     }
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct MsMembershipProof {
+pub struct RpcMsMembershipProofPrivacyPreserving {
     pub aocl_auth_paths: RpcIndexedAoclAuthPaths,
     pub target_chunks: RpcChunkDictionary,
 }
 
-impl From<MsMembershipProofPrivacyPreserving> for MsMembershipProof {
+impl RpcMsMembershipProofPrivacyPreserving {
+    /// Build the required membership proof by supplying the correct AOCL leaf
+    /// index to extract the right MMR authentication path and the missing
+    /// cryptographic data.
```

Comment:
How about "extract the right mutator set membership proof from the privacy-preserving recovery data. Picks the right AOCL MMR authentication path from the provided list."?